compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -790,19 +808,42 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
   /** Subtype test for corresponding arguments in `args1`, `args2` according to
    *  variances in type parameters `tparams`.
    */
-  def isSubArgs(args1: List[Type], args2: List[Type], tparams: List[ParamInfo]): Boolean =
+  def isSubArgs(args1: List[Type], args2: List[Type], tp1: Type, tparams: List[ParamInfo]): Boolean =
     if (args1.isEmpty) args2.isEmpty
     else args2.nonEmpty && {
-      val v = tparams.head.paramVariance
-      def isSub(tp1: Type, tp2: Type) = tp2 match {
-        case tp2: TypeBounds =>
-          tp2.contains(tp1)
+      val tparam = tparams.head
+      val v = tparam.paramVariance
+
+      def compareCaptured(arg1: Type, arg2: Type): Boolean = arg1 match {
```

Comment:
Complication: Need to do capture conversion when comparing an applied type with wildcard arguments.